planner/core/task.go

```diff
@@ -925,60 +930,76 @@ func (p *basePhysicalAgg) newPartialAggregate(copToFlash bool) (partial, final P
 		finalAggFunc.RetTp = aggFunc.RetTp
 		finalAggFuncs[i] = finalAggFunc
 	}
-
 	// add group by columns
-	groupByItems := make([]expression.Expression, 0, len(p.GroupByItems))
-	for _, gbyExpr := range p.GroupByItems {
+	finalGbyItems = make([]expression.Expression, 0, len(groupByItems))
+	for _, gbyExpr := range groupByItems {
 		gbyCol := &expression.Column{
-			UniqueID: p.ctx.GetSessionVars().AllocPlanColumnID(),
+			UniqueID: sctx.GetSessionVars().AllocPlanColumnID(),
```

Comment:
I prefer to reuse the Column if the `gbyExpr` is a `*expression.Column`, could we do this? @winoros 
If we can, I will fix this in the next PR, since it may cause a lot of tests failed.